### PR TITLE
Merge remote-tracking branch 'remotes/origin/0.2/release' into 0.2/13…

### PR DESCRIPTION
**Pull request by charlesstrange2**
…9-app-page-route-protection-authorisation

Fix login and routing.

Now people can see the app wired up again...
(Next, wire in "user" page)

### CODE REVIEW

File: fixtures/fixtures.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ var Fixtures = {
 Fixtures.apply = function (mock) {
     //TODO: example regex route;
     //TODO: example route-parsed route;
-    mock.onGet('/traders/1/vouchers').reply(200,this.traderVouchers["1"]);
+    mock.onGet('/traders/1/vouchers').reply(200, this.traderVouchers["1"]);
```
Comment (katjam) [inline fixtures/fixtures.js:54]: ugh. Why the string?

File: src/main.js [hunk 2]
```diff
@@ -6,35 +6,53 @@ import VueRouter from 'vue-router';
...
+import Scan from './pages/Scan.vue';
+import Upload from './pages/Upload.vue';
+import Payment from './pages/Payment.vue';
+import Login from './pages/Login.vue';
+import User from './pages/User.vue';
 
 // Import components
-import Masthead from './components/Masthead.vue'
-import Logo from './components/Logo.vue'
-import Instructions from './components/Instructions.vue'
```
Comment (katjam) [inline src/main.js]: Why do pages need `;` and components don't?

File: src/main.js [hunk 3]
```diff
@@ -6,35 +6,53 @@ import VueRouter from 'vue-router';
...
 ];
 
-// Create the router instance and pass the 'routes' option.
+// Create the router instance and pass the 'routes' option
 const router = new VueRouter({
-    routes, // Short for routes: routes.
-    mode: 'history'
+    routes, // short for routes: routes
+    mode: 'history',
+    base: '/'
```
Comment (katjam) [inline src/main.js]: Is that always true?

File: src/main.js [hunk 4]
```diff
@@ -6,35 +6,53 @@ import VueRouter from 'vue-router';
...
+// Route Guard rules for directing users
+router.beforeEach((to, from, next) => {
+    var auth = Store.netMgr.isAuth();
+    if (!auth && to.meta.auth) {
+        next({
+            path: '/login',
+            query: {redirect: to.fullPath} // TODO : redirect for deep logins
+        });
+    } else if (auth && !to.meta.auth)  {
+        console.log("hello");
```
Comment (katjam) [inline src/main.js]: hi!

File: src/pages/Login.vue [hunk 5]
```diff
@@ -48,20 +44,9 @@
...
-             *  Watches this.auth to check for changes
-             */
-            auth: function (val) {
-                if (this.auth) {
-                    // TODO : router wiring
-                }
+                remember: true
             }
-        },
+        },,
```
Comment (katjam) [inline src/pages/Login.vue]: 2?

File: src/pages/Login.vue [hunk 6]
```diff
@@ -73,7 +58,7 @@
                 };
                 Store.authenticate(userApiCreds, function () {
                     // I don't like this here, but it's the only place it works for now.
-                    this.auth = true;
+                        this.$router.push({path: '/tap'});
```
Comment (katjam) [inline src/pages/Login.vue]: Far out!

**Review by katjam — CHANGES_REQUESTED** [review]
Nearly. Pretty good! :balloon:

File: fixtures/fixtures.js (hunk 1 above)
Comment (charlesstrange2) [reply fixtures/fixtures.js:54]: string?

Comment (katjam) [reply fixtures/fixtures.js:54]: I mean the ugly ["1"] but that's presumably because it won't accept int keys?

**Review by katjam — APPROVED** [review]
:rocket: Now or change /tap to / as we discussed in here.

File: fixtures/fixtures.js (hunk 1 above)
Comment (charlesstrange2) [reply fixtures/fixtures.js:54]: 1 in this cas eis a label key. it happens to be the zero'th element in that array. I was toying with installing a route-parsing plugin to get laravel style route param niceness. It hasn't happened yet, but the naming system stuck.

**Pull request merged**